app/scripts/modules/core/src/pipeline/details/SingleExecutionDetails.tsx

```diff
@@ -38,108 +30,155 @@ export interface ISingleExecutionRouterStateChange extends IStateChange {
+        if (lineage.includes(stateChange.toParams.executionId)) {
+          setTransitioningToAncestor(stateChange.toParams.executionId);
         }
-      },
-    );
-    this.getExecution();
-  }
+      }
+    });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [execution]);
 
-  public componentWillUnmount(): void {
-    if (this.executionScheduler) {
-      this.executionScheduler.unsubscribe();
-    }
-    if (this.executionLoader) {
-      this.executionLoader.unsubscribe();
+  // Responsible for clearing transitioningToAncestor (allow rendering main execution again)
+  useEffect(() => {
+    if (transitioningToAncestor === execution?.id) {
+      setTransitioningToAncestor('');
     }
-    this.stateChangeSuccessSubscription.unsubscribe();
-  }
+  }, [transitioningToAncestor, execution?.id]);
 
-  private showDurationsChanged = (event: React.ChangeEvent<HTMLInputElement>): void => {
+  // Responsible for propagating showDurations back to the filterModel
+  React.useEffect(() => {
+    ExecutionState.filterModel.asFilterModel.sortFilter.showDurations = showDurations;
+  }, [showDurations]);
+
+  const { result: pipelineConfigs } = useLatestPromise<IPipeline[]>(() => {
+    app.pipelineConfigs.activate();
+    return app.pipelineConfigs.ready();
+  }, []);
+  const pipelineConfig =
+    pipelineConfigs && execution && pipelineConfigs.find((p: IPipeline) => p.id === execution.pipelineConfigId);
+
+  // Responsible for propagating showDurations changes to the filterModel
+  useEffect(() => {
+    ExecutionState.filterModel.asFilterModel.sortFilter.showDurations = showDurations;
+  }, [showDurations]);
+
+  const showDurationsChanged = (event: React.ChangeEvent<HTMLInputElement>): void => {
     const checked = event.target.checked;
-    // TODO: Since we treat sortFilter like a store, we can force the setState for now
-    //       but we should eventually convert all the sortFilters to be a valid redux
-    //       (or similar) store.
-    this.state.sortFilter.showDurations = checked;
-    this.setState({ sortFilter: this.state.sortFilter });
+    setShowDurations(checked);
     ReactGA.event({ category: 'Pipelines', action: 'Toggle Durations', label: checked.toString() });
   };
 
-  private handleConfigureClicked = (e: React.MouseEvent<HTMLElement>): void => {
+  const handleConfigureClicked = (e: React.MouseEvent<HTMLElement>): void => {
     ReactGA.event({ category: 'Execution', action: 'Configuration' });
     ReactInjector.$state.go('^.pipelineConfig', {
-      application: this.props.app.name,
-      pipelineId: this.state.execution.pipelineConfigId,
+      application: app.name,
+      pipelineId: execution.pipelineConfigId,
     });
     e.stopPropagation();
   };
 
-  private rerunExecution = (execution: IExecution) => {
-    const { app } = this.props;
-    const { pipelineConfig: pipeline } = this.state;
-
+  const rerunExecution = (execution: IExecution) => {
```

Comment:
This is a bit confusing because `execution` is passed as a parameter, but `app` and `pipelineConfig` are pulled from the closure.  Can an `execution` other than the main one ever be passed to this function?   If not, maybe just pull all the values from the closure, or add the other 2 as function params.

There is also a potential race condition if `pipelineConfig` hasn't loaded yet

Reply:
I think this was just another case of me trying to keep the diff minimal for easier review. I prefer them all as params.

`pipelineConfig && () => rerunExecution(...)` will prevents the race condition, and passing it in as a param also makes that more obvious.

Thanks!